json_serialization/reader.nim

```diff
@@ -11,6 +11,7 @@ export
 type
   JsonReader* = object
     lexer*: JsonLexer
+    forwardCompatible: bool
```

Comment:
this would be better called `allowUnknown` or something like this - forward compatibility is a much stronger claim where allowing unknown fields is a necessary but not sufficient condition

Reply:
see also https://developers.google.com/protocol-buffers/docs/proto3#unknowns and https://developers.google.com/protocol-buffers/docs/proto3#updating - the same things will apply to any serialization format and we might as well use established names for this

Reply:
Alright, will be renamed later. @cheatfate has a PR in flight that's using the current name.